tests/e2e_logging.py

```diff
@@ -434,6 +439,50 @@ def test_historical_query(network, args):
     return network
 
 
+@reqs.description("Read historical receipts")
+@reqs.supports_methods("log/private", "log/private/historical_receipt")
+def test_historical_receipts(network, args):
+    if args.consensus == "bft":
+        LOG.warning("Skipping historical queries in BFT")
+        return network
+
+    if args.package == "liblogging":
+        node, _ = network.find_primary()
+        cert_path = os.path.join(node.common_dir, f"{node.local_node_id}.pem")
+        with open(cert_path) as c:
+            node_cert = load_pem_x509_certificate(
+                c.read().encode("ascii"), default_backend()
+            )
+
+        TXS_COUNT = 5
+        network.txs.issue(network, number_txs=5)
+        for idx in range(1, TXS_COUNT + 1):
+            first_msg = network.txs.priv[idx][0]
+            first_receipt = network.txs.get_receipt(
```

Comment:
As the receipt verification has been added to the `LoggingTxs` object, could we also verify the receipts for all the transactions when calling `network.txs.verify()` (which is called at the end of each end-to-end test run)? If it takes too long, perhaps just on the full test suite?

Reply:
RE: "If it takes too long" - I recently added sampling logging to `LoggingTxs.verify`, so we only verify a small sample of historical transactions rather than every one individually. So this shouldn't take too long, though we may want to extend that sampling to "only some of the keys" as well as the current "only some of the historical entries (in each key)".

Reply:
Will have a look, but I do not like the magic verification thing that's called outside testcases. I think only testcases should have assertions and infrastructure should remain neutral.

Reply:
Not going to do this now, this requires keeping track of elections, in another PR maybe.